portfolio/src/main/webapp/scriptMainPage.js

```diff
@@ -0,0 +1,52 @@
+function onClickGithub() {
+    window.open("https://github.com/thomaspr1205");
+}
+
+function onClickLinkedin() {
+    window.open("https://www.linkedin.com/in/thomaspenar");
+}
+
+function onClickProject3() {
+    document.getElementById("project3").style.display = "block";
+}
+
+async function showRandomQuestion() {
+    const responseFromServer = await fetch('/chat');
+    const questionList = await responseFromServer.json();
+
+    const questionsContainer = document.getElementById('question-container');
+    randomIndex = Math.floor(Math.random() * 9) + 0;
+    randomQuestion = questionList[randomIndex];
+    questionsContainer.innerText = randomQuestion;
+}
+
+// Accordions
+function myAccFunc(id) {
+    var x = document.getElementById(id);
+    if (x.className.indexOf("w3-show") == -1) {
+        x.className += " w3-show";
+    } else {
+        x.className = x.className.replace(" w3-show", "");
+    }
+}
+//slideshow
+var slideIndex = 1;
```

Comment:
I would keep all variables at the top of the page. This makes it easier to see all variables used for this script.